include/abb_librws/rws_client.h

```diff
@@ -571,6 +571,17 @@ class RWSClient : public POCOClient
    */
   RWSResult setMotorsOff();
 
+  /**
+   * \brief A method for setting the robot controller's speed ratio for RAPID motions (e.g. MoveJ and MoveL).
+   *
+   * Note: The ratio must be an integer [0, 100] in text format.
+   *
+   * \param ratio specifying the new ratio.
+   *
+   * \return RWSResult containing the result.
+   */
+  RWSResult setSpeedRatio(const std::string& ratio);
```

Comment:
I guess you're accepting `std::string` here to not have to `to_string(..)` without `std::to_string(..)`?

Would using a `ostringstream` not be possible?

That would at least make it possible to enforce your preconditions (ie: `>= 0` and `<= 100`).

I realise the wrapper in `RWSInterface` already does this, but I'm not really comfortable with passing `strings` around like this. It's impossible to enfoce anything.

Reply:
Reworked the set function in 02d3cea, to take `unsigned int` instead of `std::string`.